### DISCUSSION TITLE
wimboot building document

### DISCUSSION THREAD

**Discussion by clvpr** [ipxe/wimboot#22] [Q&A] [answered]
Hi, Am not able to find documentation for  building wimboot file 
Please share any link or reference for building same

**Comment by mcb30** [comment, marked as answer]
```
git clone https://github.com/ipxe/wimboot
cd wimboot/src
make
```

**Reply by clvpr** [reply]
Thanks .. when am trying to make getting error as Makefile:176: recipe for target 'efireloc' failed
Please suggest on this

**Reply by clvpr** [reply]
I am executing  make on ubuntu machine, please suggest if I am missing any packages

**Reply by mcb30** [reply]
> Thanks .. when am trying to make getting error as Makefile:176: recipe for target 'efireloc' failed
> Please suggest on this

Please show the full error message, including the preceding lines.

If you don't have any particular need to build wimboot from source, then I would suggest that you just follow the instructions at https://ipxe.org/wimboot#download_wimboot to use the prebuilt binary.

**Reply by clvpr** [reply]
Hi 
Please find error details
administrator@administrator-virtual-machine:~/Documents/wimboot-master (1)/wimboot-master/src$ make
cc -Wall -W -Werror -I/usr/include -I/usr/include -I/usr/include -idirafter . efireloc.c -L/usr/lib -L/usr/lib -L/usr/lib -lbfd -ldl -liberty -lz -Wl,--no-warn-search-mismatch -o efireloc
/usr/bin/ld: cannot find -liberty
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
Makefile:176: recipe for target 'efireloc' failed
make: *** [efireloc] Error 1
Thanks

**Reply by mcb30** [reply]
You will need to install the relevant library packages as indicated by the error message:
```
/usr/bin/ld: cannot find -liberty
/usr/bin/ld: cannot find -lz
```
If you do not already know how to solve this, then I would very strongly suggest that you should probably just be using the prebuilt wimboot binary.

**Reply by clvpr** [reply]
Thank you so much. I am able to build wimboot file after installing below package in ubuntu
$ sudo apt-get install libiberty-dev
